test/Infer/syn-shl-from-add.opt

```diff
@@ -0,0 +1,11 @@
+; REQUIRES: solver, solver-model
+
+; RUN: %souper-check %solver -reinfer-rhs -souper-exhaustive-synthesis -souper-exhaustive-synthesis-num-instructions=1 -souper-cost-model=heuristics %s > %t
```

Comment:
this isn't a great cost model test, since many different cost models would allow this optimization to go.
please try to find a place in our unit tests where you can test the cost model directly, including whether it properly deals with the costs of external uses.

Reply:
The default one (without ignoring cost) does not infer this.

Reply:
it's not a good test case and wants to be improved